### PR TITLE
fix: use pointer everywhere when passing log chunk

### PR DESCRIPTION
**Pull request by exu**
## Pull request description 

* one way to pass and create logs (was messy pointer and non-pointer) 
* passed logs stream as parameter to executors and api

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: pkg/executor/client/job.go [hunk 1]
```diff
@@ -225,6 +230,9 @@ func (c *JobExecutor) Execute(ctx context.Context, execution *testkube.Execution
 	if err != nil {
 		return result.Err(err), err
 	}
+
+	c.logsStream.Push(ctx, execution.Id, events.NewLog("created kubernetes job"))
```
Comment (vsukhin) [inline pkg/executor/client/job.go:234]: shoud it be for v2 only?

Comment (exu) [reply pkg/executor/client/job.go:234]: 👍🏻

**Review by vsukhin — COMMENTED** [review]
sounds like feature check is missing

File: pkg/executor/client/job.go (hunk 1 above)
Comment (exu) [reply pkg/executor/client/job.go:234]: yeah need to pass feature flags too :/ good point!

**Comment by exu** [comment]
> sounds like feature check is missing

addded handling of feature flags.

**Review by vsukhin — APPROVED** [review]

**Pull request merged**